internal/gen/schema.go

```diff
@@ -59,6 +59,15 @@ func (g *Generator) mkSchema(msg *protogen.Message) error {
 		}
 
 		if field.Message != nil {
+			currentDescriptorName := descriptorName(field.Desc)
```

Comment:
What does this change fix? If it some infinite recursion problem, we probably should build a `fully-qualified proto name -> JSON Schema reference` mapping.

The following change causes stack overflow even with this patch.
```diff
diff --git a/example/message.proto b/example/message.proto
--- a/example/message.proto
+++ b/example/message.proto
@@ -35,6 +35,7 @@ message GetItemsResponse {
     string name = 3 [(google.api.field_behavior) = REQUIRED];
     google.protobuf.Timestamp created_at = 4 [(google.api.field_behavior) = REQUIRED];
     repeated NestedItem nested_item = 5;
+    repeated RecursiveNestedItem r_nested_item = 6;
   }
 
   repeated Item items = 1 [(google.api.field_behavior) = REQUIRED];
@@ -43,6 +44,10 @@ message GetItemsResponse {
   NestedEnum.ItemType nested_type = 4;
 }
 
+message RecursiveNestedItem {
+  repeated GetItemsResponse.NestedItem nested_item = 1;
+}
+
 message GetItemRequest {
   string id = 1 [
     (google.api.field_behavior) = REQUIRED,
```

Reply:
Yes, it is. Mb you have some idea how fix this?

Reply:
@tdakkota please, review it again